homeassistant/components/media_player/mpd.py

```diff
@@ -47,34 +47,13 @@
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the MPD platform."""
-    daemon = config.get(CONF_HOST)
+    host = config.get(CONF_HOST)
     port = config.get(CONF_PORT)
     name = config.get(CONF_NAME)
     password = config.get(CONF_PASSWORD)
-    import mpd
 
-    # pylint: disable=no-member
-    try:
-        mpd_client = mpd.MPDClient()
-        mpd_client.connect(daemon, port)
-
-        if password is not None:
-            mpd_client.password(password)
-
-        mpd_client.close()
-        mpd_client.disconnect()
-    except socket.error:
-        _LOGGER.error("Unable to connect to MPD")
-        return False
-    except mpd.CommandError as error:
-
-        if "incorrect password" in str(error):
-            _LOGGER.error("MPD reported incorrect password")
-            return False
-        else:
-            raise
-
-    add_devices([MpdDevice(daemon, port, password, name)])
+    device = MpdDevice(host, port, password, name)
+    add_devices([device])
```

Comment:
You can use `add_devices([device], True)` for update entity before we add it into statemachine